scopes/html/html/html.main.runtime.ts

```diff
@@ -1,18 +1,107 @@
+import { TsConfigSourceFile } from 'typescript';
+import { merge } from 'lodash';
+import type { TsCompilerOptionsWithoutTsConfig } from '@teambit/typescript';
+import { BuildTask } from '@teambit/builder';
+import { Compiler } from '@teambit/compiler';
+import { PackageJsonProps } from '@teambit/pkg';
+import { VariantPolicyConfigObject } from '@teambit/dependency-resolver';
 import { MainRuntime } from '@teambit/cli';
-import { EnvsAspect, EnvsMain } from '@teambit/envs';
+import { EnvsAspect, EnvsMain, EnvTransformer } from '@teambit/envs';
 import { ReactAspect, ReactMain } from '@teambit/react';
+import { GeneratorAspect, GeneratorMain } from '@teambit/generator';
+import { htmlEnvTemplate } from './templates/html-env';
+import { htmlComponentTemplate } from './templates/html-component';
 import { HtmlAspect } from './html.aspect';
 import { HtmlEnv } from './html.env';
 
 export class HtmlMain {
+  constructor(
+    private react: ReactMain,
+
+    readonly htmlEnv: HtmlEnv,
+
+    private envs: EnvsMain
+  ){}
   static slots = [];
-  static dependencies = [EnvsAspect, ReactAspect];
+  static dependencies = [EnvsAspect, ReactAspect, GeneratorAspect];
   static runtime = MainRuntime;
-  static async provider([envs, react]: [EnvsMain, ReactMain]) {
+
+/**
+   * override the TS config of the environment.
+   */
+  overrideTsConfig: (
+    tsconfig: TsConfigSourceFile,
+    compilerOptions?: Partial<TsCompilerOptionsWithoutTsConfig>,
+    tsModule?: any
+  ) => EnvTransformer = this.react.overrideTsConfig.bind(this.react);
+
+  /**
+   * override the jest config of the environment.
+   */
+  overrideJestConfig = this.react.overrideJestConfig.bind(this.react);
+
+  /**
+   * override the env build pipeline.
+   */
+  overrideBuildPipe: (tasks: BuildTask[]) => EnvTransformer = this.react.overrideBuildPipe.bind(this.react);
+
+  /**
+   * override the env compilers list.
+   */
+  overrideCompiler: (compiler: Compiler) => EnvTransformer = this.react.overrideCompiler.bind(this.react);
+
+  /**
+   * override the env compilers tasks in the build pipe.
+   */
+  overrideCompilerTasks: (tasks: BuildTask[]) => EnvTransformer = this.react.overrideCompilerTasks.bind(this.react);
+
+  /**
+   * override the build ts config.
+   */
+  overrideBuildTsConfig: (
+    tsconfig: any,
+    compilerOptions?: Partial<TsCompilerOptionsWithoutTsConfig>
+  ) => EnvTransformer = this.react.overrideBuildTsConfig.bind(this.react);
+
+  /**
+   * override package json properties.
+   */
+  overridePackageJsonProps: (props: PackageJsonProps) => EnvTransformer = this.react.overridePackageJsonProps.bind(
+    this.react
+  );
+
+  /**
+   * @deprecated - use useWebpack
+   * override the preview config in the env.
+   */
+  overridePreviewConfig = this.react.overridePreviewConfig.bind(this.react);
```

Comment:
can't we simply remove these APIs from the HTML env? we don't have anyone using it, so i don't see why it shoudl be here with a `@deprecate`

Reply:
agree with @itaymendel about it.
Also the overrideDevServerConfig should be removed